plugins/modules/packaging/language/maven_artifact.py

```diff
@@ -614,12 +636,16 @@ def main():
             keep_name=dict(required=False, default=False, type='bool'),
             verify_checksum=dict(required=False, default='download', choices=['never', 'download', 'change', 'always']),
             checksum_alg=dict(required=False, default='md5', choices=['md5', 'sha1']),
+            unredirected_headers=dict(type='list', elements='str', required=False, default=['Authorization', 'Cookie']),
             directory_mode=dict(type='str'),
         ),
         add_file_common_args=True,
         mutually_exclusive=([('version', 'version_by_spec')])
     )
 
+    if LooseVersion(ansible_version) < LooseVersion("2.12") and module.params['unredirected_headers']:
```

Comment:
Please note that this is a breaking change on ansible-core 2.11, since `module.params['unredirected_headers']` always evaluates to `True` if the user does not explicitly pass `unredirected_headers` to the module. You need to remove the default. What you can do is add some logic that - if `LooseVersion(ansible_version) >= LooseVersion("2.12")` - sets the value which is now the default if the value is `None` (which means that the user did not supply the parameter).